rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,216 @@
+import static co.rsk.peg.whitelist.WhitelistStorageIndexKey.LOCK_UNLIMITED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.InMemoryStorage;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.peg.whitelist.constants.WhitelistConstants;
+import co.rsk.peg.whitelist.constants.WhitelistMainNetConstants;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class WhitelistStorageProviderImplTest {
+    private final WhitelistConstants whitelistConstants = WhitelistMainNetConstants.getInstance();
+    private final NetworkParameters networkParameters = whitelistConstants.getBtcParams();
+    private WhitelistStorageProvider whitelistStorageProvider;
+    private ActivationConfig.ForBlock activationConfig;
+    private StorageAccessor inMemoryStorage;
+    private Address firstBtcAddress;
+    private Address secondBtcAddress;
+
+    @BeforeEach
+    void setUp() {
+        inMemoryStorage = new InMemoryStorage();
+        whitelistStorageProvider = new WhitelistStorageProviderImpl(inMemoryStorage);
+        activationConfig = mock(ActivationConfig.ForBlock.class);
+        firstBtcAddress = BitcoinTestUtils.createP2PKHAddress(networkParameters, "firstBtcAddress");
+        secondBtcAddress = BitcoinTestUtils.createP2PKHAddress(networkParameters, "secondBtcAddress");
+    }
+
+    @Test
+    void save_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //make sure the lockWhitelist and storage is empty
+        assertEquals(0, actualLockWhitelist.getAll().size());
+
+        //Making sure there is no value saved in Storage
+        Map<Address, OneOffWhiteListEntry> oneOffWhiteListEntryMap = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertEquals(0, oneOffWhiteListEntryMap.size());
+
+        //Making sure there is no value saved in Storage
+        Map<Address, UnlimitedWhiteListEntry> unlimitedWhiteListEntryMap = getAddressFromUnlimitedWhiteListEntryInMemoryStorage();
+        assertEquals(0, unlimitedWhiteListEntryMap.size());
+    }
```

Comment:
since we are checking the storage for both keys, lets specify it:
`// should be no value saved in one off entry`
`// should be no value saved in unlimited entry`

Reply:
Done!